Allow running detached app commands

aliismayilov opened the pull request:
This is useful for long running rake tasks or [scripts](https://github.com/rails/rails/pull/52335) that can be run without having to keep an open connection to servers.

Example:
```sh
kamal app exec 'bin/rails db:seed' --detach
```

djmb commented on lib/kamal/cli/app.rb:
```diff
@@ -94,9 +94,11 @@ def details
   def exec(*cmd)
     cmd = Kamal::Utils.join_commands(cmd)
     env = options[:env]
+    detach = options[:detach]
```
`detach` is not compatible with `reuse` or `interactive` - could you raise an error if we try to combine them?

djmb reviewed (changes requested):
Hi @aliismayilov,

Sorry for the delay in getting to this! I think we just need some checks for invalid option combinations and we should be good to go!

djmb commented:
Also it looks like there are some test failures - `CliAppTest#test_exec_detach` and `CommandsAppTest#test_execute_in_new_detached_container`

aliismayilov commented:
> Hi @aliismayilov,
> Sorry for the delay in getting to this! I think we just need some checks for invalid option combinations and we should be good to go!

Thanks for looking into this! I've added an incompatibility check. See if it makes sense.


> Also it looks like there are some test failures - `CliAppTest#test_exec_detach` and `CommandsAppTest#test_execute_in_new_detached_container`

Not anymore! I must've missed them in the avalanche of log lines 😄

aliismayilov commented:
I just realized that there will be no output from a detached container in the logs. I created a separate issue: https://github.com/basecamp/kamal/issues/1154

aliismayilov commented:
@djmb I've updated the pull request with more changes to support logs inspection for detached containers. In short:
- all newly started app containers get kamal logging configuration applied
- detached containers do not have `--rm` option. This allows inspection of detached containers even after they finished running
- `kamal logs` gets a new option `container_id` to inspect logs of a specific container

I've also updated the pull request description.

djmb commented on lib/kamal/commands/app/logging.rb:
```diff
@@ -1,15 +1,15 @@
 module Kamal::Commands::App::Logging
-  def logs(version: nil, timestamps: true, since: nil, lines: nil, grep: nil, grep_options: nil)
+  def logs(container_id: nil, timestamps: true, since: nil, lines: nil, grep: nil, grep_options: nil)
```
We need the `version` option here, it's used in `Kamal::Cli::App::Boot`

aliismayilov commented:
@djmb I've rebased the PR. There shouldn't be any conflicts or failures 😬

aliismayilov commented on lib/kamal/commands/app/logging.rb:
```diff
@@ -1,18 +1,28 @@
+
+  private
+
+  def container_id_command(container_id)
```
TBH, I'm not happy about this approach, but I couldn't come up with a better approach that would allow passing literal container ID (via kamal command) and a command that outputs container ID (see: [lib/kamal/cli/app/boot.rb](https://github.com/basecamp/kamal/pull/906/files#diff-16c00ee2b4b547d9ca9cd78761f722f4f4cc56b0f0c709c61c9beab9960f08e1R94))

aliismayilov commented:
OK, I fixed the failing spec and to be sure, the CI passes on my ~~machine~~ account: https://github.com/aliismayilov/kamal/actions/runs/11990309351
